OMCompiler/Compiler/NFFrontEnd/NFDimension.mo

```diff
@@ -199,14 +199,19 @@ public
   function size
     input Dimension dim;
     output Integer size;
+  protected
+    Dimension simple = simplify(dim);
```

Comment:
What I mean was to use `Dimension.simplify` in your own code before you call `Dimension.size`, since the need to simplify the dimension before getting the size is very rare and not something we should be paying for on every call.

Reply:
Doesn't it get optimized anyway if its not of type `Dimension.EXP()`?

Maybe i am too optimistic but i thought it doesn't make a difference for any other record in `Dimension`

Reply:
No, here you're always calling it before the algorithm section of the function.